code/game/objects/effects/particle_holder.dm

```diff
@@ -12,24 +12,29 @@
 	/// See \code\__DEFINES\particles.dm
 	var/particle_flags = NONE
 
-	var/atom/parent
+	var/datum/weakref/parent_ref
 
 /obj/effect/abstract/particle_holder/Initialize(mapload, particle_path = /particles/smoke, particle_flags = NONE)
 	. = ..()
 	if(!loc)
 		stack_trace("particle holder was created with no loc!")
 		return INITIALIZE_HINT_QDEL
 	// We nullspace ourselves because some objects use their contents (e.g. storage) and some items may drop everything in their contents on deconstruct.
-	parent = loc
+	parent_ref = WEAKREF(loc)
 	loc = null
 
 	// Mouse opacity can get set to opaque by some objects when placed into the object's contents (storage containers).
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	src.particle_flags = particle_flags
 	particles = new particle_path()
 	// /atom doesn't have vis_contents, /turf and /atom/movable do
-	var/atom/movable/lie_about_areas = parent
-	lie_about_areas.vis_contents += src
+	var/atom/movable/parent = parent_ref.resolve()
+	if(isnull(parent))
+		parent_ref = null
+		stack_trace("particle holder was created without a valid parent!")
+		return INITIALIZE_HINT_QDEL
+
+	parent.vis_contents += src
 	RegisterSignal(parent, COMSIG_QDELETING, PROC_REF(parent_deleted))
```

Comment:
(see here)

Reply:
Particle holders only exist if their parent exists, so we want to know when the parent is qdeleted so we too can qdel. 
Weakrefs aren't good for that, since they can delete whenever they want without us reacting.

Reply:
To be fair weakrefs isn't a byond thing but yeah. 

This does WORK as a weakref but we're already registering the qdeleting signals on the atom so there's zero reason to use it over a normal reference